src/main/java/com/epam/test_generator/entities/Token.java

```diff
@@ -24,44 +25,50 @@
     @NotNull
     private String token;
 
-    @NotNull
-    private Date expiryDate;
-
     @OneToOne
     private User user;
 
-    public User getUser() {
-        return user;
-    }
-
-    public void setUser(User user) {
-        this.user = user;
-    }
-
-    public Token() {
+    @NotNull
+    private Date expiryDate;
 
+    public Token(Integer minutes) {
+        this.token = UUID.randomUUID().toString();
```

Comment:
IMO static builder would be more clean way to express what this piece of code performs.

    public static Token withExpiryDuration(int minutes) {
        Token result = new Token();
        result.token = UUID.randomUUID().toString();
        result.expiryDate = Date.from(Instant.now().plus(minutes, ChronoUnit.MINUTES));
        return result;
    }

Reply:
Done